### PR TITLE
[SPARK-12217] [ML] Document invalid handling for StringIndexer

### PR DESCRIPTION
**Pull request by BenFradet**
Added a paragraph regarding StringIndexer#setHandleInvalid to the ml-features documentation.

I wonder if I should also add a snippet to the code example, input welcome.

### CODE REVIEW

**Comment by BenFradet** [comment]
Pinging @holdenk and @jkbradley

**Comment by holdenk** [comment]
That looks good to me, I don't think a full code example is necessary.

File: docs/ml-features.md [hunk 1]
```diff
@@ -459,6 +459,42 @@ column, we should get the following:
 "a" gets index `0` because it is the most frequent, followed by "c" with index `1` and "b" with
 index `2`.
 
+Additionaly, there are two strategies regarding how `StringIndexer` will handle
+unseen labels when you have set up a `StringIndexer` on a dataset which you want
```
Comment (jkbradley) [inline docs/ml-features.md]: "set up" --> "fit"
"on a dataset which you want to reuse on another" --> "on one dataset and then use it to transform another dataset"

**Comment by jkbradley** [comment]
@BenFradet  Thanks!  I agree you didn't have to write a full example, but it's nice that it explains it very clearly, so I'd keep it.

I just had small phrasing comments.

**Comment by BenFradet** [comment]
@jkbradley thanks for the comments.

**Comment by jkbradley** [comment]
LGTM pending tests
Thanks!

**Comment by jkbradley** [comment]
Merging with master and branch-1.6

**Pull request closed without merging**